### ISSUE TITLE
The repeat group with additionall group inside is skipped when I navigate backward

### ISSUE THREAD

**Issue by grzesiek2010** [opendatakit/javarosa#218]
#### Software versions 
JavaRosa v2.6.0

#### Problem description
The issue has been originally reported in [the Collect repository](https://github.com/opendatakit/collect/issues/1582) but after my investigation, I'm pretty sure it's on the Javarosa side.

The problem occurs when I want to use a repeat group with an additional group inside. That group can't display all questions on one screen (in that case everything looks ok). Then when I want to return to questions in such a group those questions are skipped and I'm moved to the question which is before the repeat group.

#### Steps to reproduce the problem
1. Use the attached form
2. Navigate Foward when you are on the first question
3. Add one group.
4. Navigate Foward when you are on the first question in the group
5. Navigate Foward when you are on the second question in the group
6. Don't add another group
7. Navigate Backward

#### Expected behavior
I should be moved to the last question in the repeat group.

#### Other information 
This form is a good example of the issue:
[rp1.xml.txt](https://github.com/opendatakit/javarosa/files/1449497/rp1.xml.txt)

Using the attached form I investigated [indexes the Javarosa](https://github.com/opendatakit/javarosa/blob/master/src/org/javarosa/form/api/FormEntryController.java#L200) uses when I navigate in the form and here is a small example:

```
0, // Text - navigate forward
1_0, // Prompt new repeat - click on the add button
1_0, 0,
1_0, 0, 0, // Number - navigate forward
1_0, 0, 1, // Barcode - navigate forward
1_1 	// Prompt new repeat - click on the don't add button
-1 	// End of form - navigate backward
1_1 	// this is skipped automatically what is ok
	// QUESTION IS SKIPPED !!!
	// QUESTION IS SKIPPED !!!
1_0, 0,
1_0, 
0, 
```

Everything we do on the Collect side is in [stepToPreviousScreenEvent()](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/logic/FormController.java#L534) method. Looking at this part of code using the same form I can see that when I navigate backward (step 7):

[the returned event](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/logic/FormController.java#L537) is 2 - EVENT_PROMPT_NEW_REPEAT which is ok and we try to skip this event [here](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/logic/FormController.java#L545) but after that, we are moved back and all questions are skipped.

**Comment by grzesiek2010** [comment]
@mdudzinski maybe you could take a look at it?

**Comment by mdudzinski** [comment]
@grzesiek2010 Sure, I'll take a look!
@opendatakit-bot claim

**Comment by mdudzinski** [comment]
@grzesiek2010 You are right, the problem is in `decrementIndex(FormIndex index)` method. Thanks for the initial findings!

**Comment by mdudzinski** [comment]
@grzesiek2010 
It turns out that it will work if the nested group has the `ref` attribute which will point to the corresponding element in the instance. 
Slightly updated form: 
[rp1-group-with-ref-attribute.xml.txt](https://github.com/opendatakit/javarosa/files/1481905/rp1-group-with-ref-attribute.xml.txt)

However, it should be possible to use the `nodeset` attribute instead of the `ref` (according to the specs -> https://opendatakit.github.io/xforms-spec/#body-attributes) but this attribute is currently ignored for the `<group>` elements. There is a fix on the way.

**Comment by grzesiek2010** [comment]
@mdudzinski Nice!
I used for builder https://build.opendatakit.org/ and in that case the `ref` is not added but it works well if I use the XLSForm converter http://opendatakit.org/xiframe/

So looks like it's also a bug on the builder's side too. @lognaturel what do you think?

**Comment by lognaturel** [comment]
Nice work, detectives. 🔎 

Comment on #224. It does seem like Build should include the `ref`.

**Comment by lognaturel** [comment]
Forms created by Build will still have this problem for now but it's filed.
